pkg/proxy/proxy.go

```diff
@@ -454,7 +454,8 @@ func (p *Proxy) CreateOrUpdateRedirect(l4 *policy.L4Filter, id string, localEndp
 	pp := getProxyPort(l4.L7Parser, l4.Ingress)
 	if pp == nil {
 		err = proxyNotFoundError(l4.L7Parser, l4.Ingress)
-		return
+		revertFunc()
```

Comment:
The commit message does not seem to correspond what this is doing. Here, it seems you are running `revertFunc` if `err` is or is not nil

Reply:
`pp == nil` is the error condition here, and the `err = ` line sets the `err` return value. We call `revertFunc`.

Reply:
ah right. I didn't read the function name "proxyNotFoundError" so I didn't realize that `proxyNotFoundError` was actually creating an error.